### ISSUE TITLE
Only 9 posts load on All+Hot feed

### ISSUE THREAD

**Issue by gon555551** [thunder-app/thunder#502]
**Description**  
Only 9 posts load when the feed is set to All+Hot. 

**How to Reproduce**  
Set feed to All+Hot. I don't know what else may be causing this.

**Expected Behavior**  
It should load way more posts.

**Screenshots**  
![Showing my feed on All+Hot, where it says I've reached the bottom after only 9 posts loaded.](https://i.postimg.cc/HnJGh1N4/Screenshot-2023-07-30-23-11-07-811.jpg)

**Device & App Version:**  
 - Device: TCL20+
 - OS: Android 11
 - Version: 0.2.1+13

**Additional Context**  
It doesn't happen all the time, but it does happen rather frequently. 

I've tried different accounts on different instances and the problem remains.  
I've tried closing the app and opening it again, the problem remains. 

I don't notice any other issues, and other feed settings (Local, Home, Active, New, Top, etc) all work fine. It's just on All+Hot. 

My connection seems fine, and I can't find any other factor that may be causing this.

**Comment by CTalvio** [comment]
I also encountered this, but only once. If it happens again, could you make sure other apps and the lemmy webUI do load more posts, in order to rule out the issue lying with the lemmy server, and not thunder.

**Comment by gon555551** [comment]
Haven't checked other Android apps, but web does load more posts.

**Comment by hjiangsu** [comment]
Thats a weird one, I'll see if I can reproduce it on my end but seems like it'll be a harder one to track down since all these options should follow the same logic for fetching posts

**Comment by hjiangsu** [comment]
This could be related to #534

**Comment by hjiangsu** [comment]
I'll mark this as fixed, but do let me know if this happens again!

**Comment by gon555551** [comment]
After I updated to +14 it's happening again!! So weird...

**Comment by hjiangsu** [comment]
Do you happen to have "Show Link Preview" disabled in the settings?

**Comment by gon555551** [comment]
It's enabled.

![Enabled.](https://i.postimg.cc/bJSCKx8z/Screenshot-2023-08-08-14-35-10-444.jpg)

**Comment by hjiangsu** [comment]
This is weird. I still can't reproduce the issue. When you load All+Hot, are you able to scroll on the feed?

Could you also try to clear out your preferences in Settings -> Debug -> Delete Local Preferences? This will reset your preferences so just a heads up

**Comment by hjiangsu** [comment]
Alright sounds good! Yeah, I'll continue to look out for it but it's hard to track down this one unfortunately 😅

**Comment by hjiangsu** [comment]
I think I've figured out the issue. It happens when you have NSFW posts which are being loaded in your feed (All+Hot), but you have the NSFW content setting off. The NSFW posts are still being fetched but they are filtered from your view. This is why it feels like it randomly happens, because this only really happens on the first load (hopefully)

When this happens, the logic that determines if there are more posts to load fail, and prevents you from loading more posts. I've added the fix, and this will come in the next nightly version!
